modules/lib/lib-content/src/main/resources/lib/xp/content.ts

```diff
@@ -1008,7 +1008,7 @@ interface QueryContentHandler {
  *
  * @returns {object} Result of query.
  */
-export function query<Data = Record<string, unknown>, Type extends string = string>(params: QueryContentParams): ContentsResult<Data, Type> {
+export function query<DataAndType extends DataAndTypeTuple = readonly [data: Record<string, unknown>, type: string]>(params: QueryContentParams): ContentsResult<DataAndType> {
```

Comment:
Are you sure we want a Tuple here? It always return a single Content, so it might be consistent to keep <Data, Type>, like with functions that operate on with single value.

Reply:
This is `query()` which takes in `contentTypes` as a parameter, so the results can contain different content types.

See `getContent()` for the more interesting answer :)